### DISCUSSION TITLE
Creating sensor for speed, convert to km/h

### DISCUSSION THREAD

**Discussion by bartowski1182** [alandtse/tesla#578] [Q&A] [answered]
With the new change to the MQTT integration to convert to mph, what's the best way for me to set up a sensor that will convert it back to km/h for me?

this is the template I'm currently using doesn't seem to be working correctly:

- sensor:
  - name: "Tesla Speed"
    device_class: speed
    unit_of_measurement: "km/h"
    state: >
      {% if state_attr('device_tracker.thunderfury_location_tracker', 'speed') == None %}
        0
      {% else %}
        {{ state_attr('device_tracker.thunderfury_location_tracker', 'speed')  | float * 1.609 | round(0) }}
      {% endif %}

it's displaying speeds far above what the car is actually driving (as I watch it in both the Tesla app and in TeslaMate), 99 km/h is being converted to 124 km/h

**Comment by bartowski1182** [comment]
Apparently my MQTT broker wasn't running, so it's possible the integration just happened to pull when the speed spiked and i didn't realize, will monitor it and update. Still interested in whether or not this sensor is the best way to do it

**Comment by bartowski1182** [comment, marked as answer]
Okay so I was stupid, and writing it this way causes 1.609 to be rounded up to 2

the actual way to write it is with an extra bracket:
{{ (state_attr('device_tracker.thunderfury_location_tracker', 'speed') | float * 1.609) | round }}

closing as solution

**Reply by alandtse** [reply]
Glad to see you resolved it. The wiki may be a good place to reference this discussion if you're willing to update it.
